api/src/db/migrations/v138_multi_license_support.test.ts

```diff
@@ -0,0 +1,260 @@
+    expect(updatedLicenseDetails?.licenseStatus).toBe(initialBusiness.licenseData?.status);
+    expect(updatedLicenseDetails?.expirationDateISO).toBe(initialBusiness.licenseData?.expirationISO);
+    expect(updatedLicenseDetails?.lastUpdatedISO).toBe(initialBusiness.licenseData?.lastUpdatedISO);
+    expect(updatedLicenseDetails?.checklistItems).toEqual(initialBusiness.licenseData?.items);
+    expect(updatedLicenseDetails?.hasError).toBeFalsy();
+  });
+
+  it("correctly updates register-consumer-affairs taskprogress key to register-home-contractor when license data is undefined", () => {
+    const industryId = "home-contractor";
+    const taskId = "register-home-contractor";
+
+    const initialBusiness = generatev137Business({
+      taskProgress: initialTaskProgress,
+      licenseData: undefined,
+      profileData: generatev137ProfileData({ industryId: industryId }),
+    });
+
+    // TODO my check, delete
+    expect(LookupIndustryById(industryId).id).toBeTruthy();
+
+    const updatedBusiness = migrate_v137Business_to_v138Business(initialBusiness);
+
+    expect(updatedBusiness.taskProgress).toEqual({
+      "non-license-task-progress": "COMPLETED",
+      [taskId]: "COMPLETED",
+    });
+
+    expect(updatedBusiness.licenseData).toBeUndefined();
+  });
+
+  it("correctly updates the license data and changes appraiser-license taskprogress key to appraiser-company-register", () => {
+    const industryId = "real-estate-appraisals";
+    const licenseName = "Real Estate Appraisers-Appraisal Management Company";
+    const taskId = "appraiser-company-register";
+
+    const initialBusiness = generatev137Business({
+      taskProgress: initialTaskProgress,
+      licenseData: initialLicenseData,
+      profileData: generatev137ProfileData({ industryId: industryId }),
+    });
+    // TODO my check, delete
+    expect(LookupIndustryById(industryId).id).toBeTruthy();
+
+    const updatedBusiness = migrate_v137Business_to_v138Business(initialBusiness);
+
+    expect(updatedBusiness.taskProgress).toEqual({
+      "non-license-task-progress": "COMPLETED",
+      [taskId]: "COMPLETED",
+    });
+
+    expect(updatedBusiness.licenseData?.lastUpdatedISO).toEqual(initialBusiness.licenseData?.lastUpdatedISO);
+
+    const updatedLicenseDetails = updatedBusiness.licenseData?.licenses?.[licenseName as v138LicenseName];
+    expect(updatedLicenseDetails?.nameAndAddress).toEqual(initialBusiness.licenseData?.nameAndAddress);
+    expect(updatedLicenseDetails?.licenseStatus).toBe(initialBusiness.licenseData?.status);
+    expect(updatedLicenseDetails?.expirationDateISO).toBe(initialBusiness.licenseData?.expirationISO);
+    expect(updatedLicenseDetails?.lastUpdatedISO).toBe(initialBusiness.licenseData?.lastUpdatedISO);
+    expect(updatedLicenseDetails?.checklistItems).toEqual(initialBusiness.licenseData?.items);
+    expect(updatedLicenseDetails?.hasError).toBeFalsy();
```

Comment:
[dust] Just to clarify, are you hoping to check that this is either false or undefined? If you are trying to check that this is false, you should use toBe(false)

Reply:
same with the toBeTruthy() on 241

Reply:
Fixed